squbs-unicomplex/src/test/scala/org/squbs/stream/PerpetualStreamMatValueSpec.scala

```diff
@@ -127,18 +132,18 @@ object PerpetualStreamMatValueSpecHelper {
   val timeoutDuration = 1.second
   implicit val timeout = Timeout(timeoutDuration)
 
-  def useSystem[PC <: PerpStream[_]](perpStream: Class[PC])(fn: Try[ActorRef] => Unit): Unit = {
-    implicit val actorSystem = ActorSystem()
-    implicit val materializer = ActorMaterializer()
+  def useSystem[PC <: PerpStream[_] : ClassTag](fn: Try[ActorRef] => Unit)
+                                               (implicit system: ActorSystem, mat: Materializer): Unit = {
 
-    val perpRef = actorSystem.actorOf(Props(perpStream))
-    val someRef = actorSystem.actorOf(Props(new SomeActor(perpRef)))
+    val perpRef = system.actorOf(Props[PC])
+    val someRef = system.actorOf(Props(new SomeActor(perpRef)))
```

Comment:
I'm reverting this line to what is was before. After double-checking the `Props` code, this is the correct way to create this actor. And no, it does not create the actor at the creation of props (as was the issue earlier). It takes this `by-name` parameter and executes it when the actor is indeed created. Type inference allows us not to have to pass a type parameter here. And the `classTag` is generated by the compiler to full in the type information. Ingenious!